components/movementsensor/imuwit/imu.go

```diff
@@ -204,13 +205,28 @@ func NewWit(
 	utils.PanicCapturingGo(func() {
 		defer utils.UncheckedErrorFunc(port.Close)
 		defer i.activeBackgroundWorkers.Done()
-
+		timer := time.NewTicker(time.Duration(100 * float64(time.Millisecond)))
+		defer timer.Stop()
 		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			select {
+			case <-ctx.Done():
+				return
+			case <-timer.C:
+			}
+
 			if ctx.Err() != nil {
 				return
 			}
 
 			line, err := portReader.ReadString('U')
+			if err != nil {
+				i.logger.Errorf("err in line %#v", err)
```

Comment:
perhaps this should be worded like "error reading from port, %#v" or something like that?

Reply:
done